waiter/src/waiter/scheduler/marathon.clj

```diff
@@ -395,11 +419,143 @@
   (service-id->state [_ service-id]
     {:failed-instances (service-id->failed-instances service-id->failed-instances-transient-store service-id)
      :killed-instances (scheduler/service-id->killed-instances service-id)
-     :kill-info (get @service-id->kill-info-store service-id)})
+     :kill-info (get @service-id->kill-info-store service-id)
+     :out-of-sync-state (get @service-id->out-of-sync-state-store service-id)})
 
   (state [_]
     {:service-id->failed-instances-transient-store @service-id->failed-instances-transient-store
-     :service-id->kill-info-store @service-id->kill-info-store}))
+     :service-id->kill-info-store @service-id->kill-info-store
+     :service-id->out-of-sync-state-store @service-id->out-of-sync-state-store}))
+
+(defn- get-apps-with-deployments
+  "Retrieves the apps with the deployment info embedded."
+  [{:keys [marathon-api is-waiter-app?-fn]}]
+  (get-apps marathon-api is-waiter-app?-fn {"embed" ["apps.deployments" "apps.tasks"]}))
+
+(defn retrieve-service-id->out-of-sync-state
+  "Retrieves the service-id->out-of-sync-state."
+  [{:keys [service-id->out-of-sync-state-store]}]
+  (deref service-id->out-of-sync-state-store))
+
+(defn- reset-service-id->out-of-sync-state!
```

Comment:
If `process-out-of-sync-services!` returned the new value rather than doing the `reset!` inline, then `reset-service-id->out-of-sync-state!` would similarly be called in only one function, and it would probably be simpler to understand the inlined `reset!` rather than scrolling around to find this separate definition.